dtests/src/test/java/io/snappydata/hydra/cluster/SnappyTest.java

```diff
@@ -2238,9 +2247,18 @@ public void executeSnappyJob(Vector jobClassNames, String logFileName, String
         logFile = new File(dest);
         snappyTest.executeProcess(pb, logFile);
         pb = new ProcessBuilder("/bin/bash", "-c", curlCommand2);
-        snappyTest.executeProcess(pb, logFile);
+        int status = snappyTest.executeProcess(pb, logFile);
       }
       boolean retry = snappyTest.getSnappyJobsStatus(snappyJobScript, logFile, leadPort);
+      if(!checkJobStatus(getJobIDs(logFile)) && !cycleVms){
+        if(SnappyPrms.isReRunWithDebugEnabled())
+        {
+          HydraTask_changeLogLevel();
+          HydraTask_executeSnappyJob();
+        }
+        throw new TestException
+            ("Snappy job execution has failed. Please check the logs.");
```

Comment:
Log the exception instead of failing the test there itself. We need to report the failure for each case at the end of the test after verifying all scenarios.